### PR TITLE
2711: Add apks to GitHub releases

### PR DESCRIPTION
**Pull request by ztefanie**
### Short description
Added native assets (.apks and .ipas) to the github releases.

### Proposed changes
- Remove the "beta" from the name of a release
- Added pre-release tag for beta release
- Remove pre-release tag after promotion
- always mark the android release as latest, so we can link easily to our apks, with github/digitalfabrik/integreat-app/releases/latest
- Integreat can be downloaded by: https://github.com/digitalfabrik/integreat-app/releases/latest/download/integreat.apk

More prominent linking to the apks will be done here: https://openproject.tuerantuer.org/projects/infra/work_packages/3176/activity

### Side effects
- hopefully none, may break something else in the CI

### Resolved issues

Fixes: #2711

---

### CODE REVIEW

File: tools/github-release.ts [hunk 1]
```diff
@@ -8,43 +8,35 @@ type Options = {
...
     owner,
     repo,
     tag_name: tagId({ versionName: newVersionName, platform }),
+    prerelease: prerelease === 'true',
+    make_latest: platform === 'android' ? 'true' : 'false',
     name: releaseName,
     body,
   })
+
+  console.log(release.data.id)
```
Comment (LeandraH) [inline tools/github-release.ts:39]: ❓  Is this still test code? If not, maybe elaborate in the message what that id is?

**Review by LeandraH — APPROVED** [review]
👍 Thanks for figuring this out!
❓ It looks reasonable to me but I don't quite understand the pre-release removal for web. Is that just to keep it uniform?

File: .circleci/src/jobs/notify_ios.yml [hunk 2]
```diff
@@ -18,7 +18,11 @@ steps:
       working_directory: ~/attached_workspace
   - run:
       name: Create github release
-      command: yarn github-release create ios ${NEW_VERSION_NAME} ${NEW_VERSION_CODE} --deliverino-private-key ${DELIVERINO_PRIVATE_KEY} --owner ${CIRCLE_PROJECT_USERNAME} --repo ${CIRCLE_PROJECT_REPONAME} --release-notes "$(yarn --silent manage-metadata parse-release-notes --ios --language en)"<<^ parameters.production_delivery >> --beta-release<</ parameters.production_delivery >> --download-links "${IOS_ARTIFACT_URLS}"
+      command: IOS_RELEASE_ID=$(yarn --silent github-release create ios ${NEW_VERSION_NAME} ${NEW_VERSION_CODE} --prerelease !<< parameters.production_delivery >> --deliverino-private-key ${DELIVERINO_PRIVATE_KEY} --owner ${CIRCLE_PROJECT_USERNAME} --repo ${CIRCLE_PROJECT_REPONAME} --release-notes "$(yarn --silent manage-metadata parse-release-notes --ios --language en)")
+      working_directory: tools
+  - run:
+      name: Upload asset to github release
```
Comment (steffenkleinle) [inline .circleci/src/jobs/notify_ios.yml:24]: :upside_down_face: 
```suggestion
      name: Upload ipas to github release
```

File: .circleci/src/jobs/notify_android.yml [hunk 3]
```diff
@@ -18,7 +18,11 @@ steps:
       working_directory: ~/attached_workspace
   - run:
       name: Create github release
-      command: yarn github-release create android ${NEW_VERSION_NAME} ${NEW_VERSION_CODE} --deliverino-private-key ${DELIVERINO_PRIVATE_KEY} --owner ${CIRCLE_PROJECT_USERNAME} --repo ${CIRCLE_PROJECT_REPONAME} --release-notes "$(yarn --silent manage-metadata parse-release-notes --android --language en)"<<^ parameters.production_delivery >> --beta-release<</ parameters.production_delivery >> --download-links "${ANDROID_ARTIFACT_URLS}"
+      command: echo "export ANDROID_RELEASE_ID='$(yarn --silent github-release create android ${NEW_VERSION_NAME} ${NEW_VERSION_CODE} --prerelease !<< parameters.production_delivery >> --deliverino-private-key ${DELIVERINO_PRIVATE_KEY} --owner ${CIRCLE_PROJECT_USERNAME} --repo ${CIRCLE_PROJECT_REPONAME} --release-notes "$(yarn --silent manage-metadata parse-release-notes --android --language en)")'" >> ${BASH_ENV}
+      working_directory: tools
+  - run:
+      name: Upload asset to github release
```
Comment (steffenkleinle) [inline .circleci/src/jobs/notify_android.yml:24]: :upside_down_face: 
```suggestion
      name: Upload apks to github release
```

File: tools/github-release-asset.ts [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+
+type Options = {
+  deliverinoPrivateKey: string
+  owner: string
+  repo: string
+  releaseId: number
+  files: string
+}
+
+const uploadAsset = async ({ deliverinoPrivateKey, owner, repo, releaseId, files }: Options) => {
```
Comment (steffenkleinle) [inline tools/github-release-asset.ts:14]: :upside_down_face: 
```suggestion
const uploadAssets = async ({ deliverinoPrivateKey, owner, repo, releaseId, files }: Options) => {
```
Also you are inconsistent between `asset` and `files`. Perhaps you could adjust that.

**Review by steffenkleinle — APPROVED** [review]
Looking good! :tada:

**Comment by steffenkleinle** [comment]
>❓ It looks reasonable to me but I don't quite understand the pre-release removal for web. Is that just to keep it uniform?

How do you mean that? On web we have the same workflow, first a delivery to beta and one week later a promotion to production. Therefore prereleases make sense here as well.

File: tools/github-release.ts (hunk 1 above)
Comment (ztefanie) [reply tools/github-release.ts:39]: No it is not. This is the output of the command, which returns this id.

File: tools/github-release-asset.ts (hunk 4 above)
Comment (ztefanie) [reply tools/github-release-asset.ts:14]: I use both wordings, but they are slightly different:
files: These are the files on the disk (of circle ci)
assets: These are the assets of the github release.
It think it makes sense to keep both wordings as they are different. Think of it like an attachment of an email, where you attach a file.

**Pull request merged**